ResponseDetective/Sources/URLEncodedBodyDeserializer.swift

```diff
@@ -0,0 +1,42 @@
+//
+// URLEncodedBodyDeserializer.swift
+//
+// Copyright © 2016-2017 Netguru Sp. z o.o. All rights reserved.
+// Licensed under the MIT License.
+//
+
+import Foundation
+
+/// Deserializes URL-encoded bodies.
+@objc(RDTURLEncodedBodyDeserializer) public final class URLEncodedBodyDeserializer: NSObject, BodyDeserializer {
+
+	// MARK: BodyDeserializer
+
+	/// Deserializes JSON data into a pretty-printed string.
+	public func deserialize(body: Data) -> String? {
+
+		guard let string = String(data: body, encoding: .utf8) else {
+			return nil
+		}
+
+		if #available(macOS 10.10, tvOS 10.0, *) {
+
+			let stringFromQueryItems: ([URLQueryItem]) -> String = {
+				$0.map { "\($0.name): \($0.value ?? "nil")" }.joined(separator: "\n")
+			}
+
+			if let items = URLComponents(string: string)?.queryItems {
```

Comment:
Tests pass so it doesn't. 😉 

Apple documentation has been lying about availability from day one, see [`-[NSArray init]`](https://developer.apple.com/reference/foundation/nsarray/1414315-init) for example. It says it's available since... iOS 7 and macOS 10.9. :trollface: 

That availability is misleading since the symbols such as `NSArray` or `URLComponents` are introduced in _Swift releases_, not _Foundation releases_. But since Apple bundles Swift and SDK releases, symbols that have been introduced in Swift 3, are always marked as available since SDK versions, such as iOS 10, macOS 10.12, etc.

`URLComponents` has been introduced in Swift 3 (along with macOS 10.12), yes. But as it's only a bridging type and since Swift's stdlib is bundled with the app, the only thing that matters is the underlying `NSURLComponents` class. And that's available for a long time.

If you cmd+click `URLComponents`, you'll see no `@availability` attributes.

The check you commented above is for `URLComponents.queryItems`, which _seriously_ was introduced in macOS 10.10 (it has an `@availability` macro).